ruby/ext/google/protobuf_c/encode_decode.c

```diff
@@ -645,21 +651,19 @@ VALUE Message_decode(VALUE klass, VALUE data) {
 
   const upb_pbdecodermethod* method = msgdef_decodermethod(desc);
   const upb_handlers* h = upb_pbdecodermethod_desthandlers(method);
-  upb_pbdecoder decoder;
-  upb_sink sink;
-  upb_status status = UPB_STATUS_INIT;
+  upb_env e;
+  upb_env_init(&e);
```

Comment:
Pushed an update now that the upb PR is in. Using stack-allocated storage, we're back to ~parity -- 0.508s baseline vs. 0.509s PR for the above benchmark (trials=10).